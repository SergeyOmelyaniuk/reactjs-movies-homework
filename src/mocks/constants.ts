import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
