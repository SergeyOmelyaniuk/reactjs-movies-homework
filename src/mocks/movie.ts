import { MovieDetails } from './../types';

export const movieDescription: MovieDetails = {
	poster_path: 'film_1.jpg',
	genres: [{ id: 1, name: 'first genre' }],
	title: 'Красотка на взводе',
	release_date: '2021-05-26',
	vote_average: 68,
	revenue: 120000,
	runtime: 120,
	overview:
		'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
};
