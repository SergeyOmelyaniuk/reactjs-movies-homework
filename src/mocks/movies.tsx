import { Movie } from '../types';

export const movies: Movie[] = [
	{
		id: 1,
		title: 'Красотка на взводе',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_1.jpg',
		vote_average: 68,
		popularity: 0.5,
		revenue: 120000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [
			{
				name: 'Kate Beckinsale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_1.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Lindy',
			},
			{
				name: 'Stanley Tucci',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_2.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Dr. Ivan Munchin',
			},
			{
				name: 'Bobby Cannavale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_3.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Vicars',
			},
			{
				name: 'Jai Courtney',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_4.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Justin',
			},
			{
				name: 'Laverne Cox',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_5.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Nevin',
			},
			{
				name: 'David Bradley',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_6.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Gareth Fizel',
			},
		],
	},
	{
		id: 2,
		title: 'Сокровище Счастливчика',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_2.jpg',
		vote_average: 72,
		popularity: 0.5,
		revenue: 230000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 3,
		title: 'Сталинград',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_3.jpg',
		vote_average: 76,
		popularity: 0.5,
		revenue: 240000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 4,
		title: 'Люди в черном: Интернэшнл',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_4.jpg',
		vote_average: 69,
		popularity: 0.5,
		revenue: 136000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 5,
		title: 'Значит, война',
		release_date: '2021-05-26',
		genre: ['Боевик'],
		img: 'film_5.jpg',
		vote_average: 96,
		popularity: 0.5,
		revenue: 226000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
];
