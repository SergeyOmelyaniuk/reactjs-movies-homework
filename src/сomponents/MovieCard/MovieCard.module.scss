@import '../../scss/variables.scss';

.movieCard {
	display: inline-block;
	width: 253px;
	cursor: pointer;
	&:hover .preview {
		display: block;
	}
	&:hover .title {
		color: $colorWhite;
	}
	&:hover .overlay {
		position: absolute;
	}
	@media (max-width: 1600px) {
		width: 210px;
	}
}

.overlay {
	top: 0;
	bottom: 0;
	left: 0;
	right: 0;
	background: $colorOverlay;
	border-radius: 5px;
}

.wrapImage {
	position: relative;
	max-width: 253px;
	width: 100%;
	height: 366px;
	@media (max-width: 1600px) {
		max-width: 210px;
		height: 304px;
	}
}

.image {
	height: 100%;
	width: 100%;
	display: block;
	border-radius: 5px;
}

.rating {
	display: flex;
	justify-content: center;
	align-items: center;
	position: absolute;
	left: -8px;
	top: 6px;
	background-color: $colorGreen;
	color: $colorWhite;
	width: 42px;
	height: 24px;
	border-radius: 3px;
	font-size: 14px;
	@media (max-width: 1600px) {
		left: -7px;
		top: 5px;
		width: 35px;
		height: 20px;
		font-size: 12px;
	}
}

.preview {
	position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
	width: 62px;
	height: 62px;
	background: $colorPreview url('../../assets/images/icon-play.png') no-repeat
		center;
	border-radius: 50%;
	display: none;
	@media (max-width: 1600px) {
		width: 51px;
		height: 51px;
	}
}

.content {
	color: $colorGrey;
}

.title {
	font-size: 20px;
	margin-top: 15px;
	margin-bottom: 9px;
	line-height: 25px;
	@media (max-width: 1600px) {
		font-size: 17px;
		margin-top: 12px;
		margin-bottom: 7px;
		line-height: 20px;
	}
}

.genre {
	line-height: 1.5;
	@media (max-width: 1600px) {
		line-height: 1.2;
	}
	& span {
		margin-right: 10px;
		font-size: 14px;
		display: inline-block;
		color: $colorGreyDark;
		@media (max-width: 1600px) {
			margin-right: 8px;
			font-size: 11px;
		}
	}
}
