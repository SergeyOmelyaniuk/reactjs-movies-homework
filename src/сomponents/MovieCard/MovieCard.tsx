import React from 'react';
import { Movie } from '../../types';
import styles from './MovieCard.module.scss';

const MovieCard = ({ movie }: { movie: Movie }) => (
	<div className={styles.movieCard}>
		<div className={styles.wrapImage}>
			<div className={styles.overlay}></div>
			<img
				className={styles.image}
				src={`https://image.tmdb.org/t/p/w500/${movie.poster_path}`}
				alt=''
			/>
			<span className={styles.rating}>{movie.vote_average}</span>
			<div className={styles.preview}></div>
		</div>
		<div className={styles.content}>
			<h2 className={styles.title}>{movie.title}</h2>
			<p className={styles.genre}>
				{movie.genre_ids.map((item: string) => (
					<span key={item}>{item}</span>
				))}
			</p>
		</div>
	</div>
);

export default MovieCard;
