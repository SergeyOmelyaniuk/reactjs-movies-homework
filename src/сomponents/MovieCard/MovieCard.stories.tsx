import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import MovieCard from './MovieCard';
import { movies } from '../../mocks/movies';

export default {
	title: 'Components/MovieCard',
	component: MovieCard,
	argTypes: {},
} as ComponentMeta<typeof MovieCard>;

const Template: ComponentStory<typeof MovieCard> = (args) => (
	<div
		style={{
			width: '300px',
			height: '500px',
			backgroundColor: '#202020',
			display: 'flex',
			justifyContent: 'center',
			alignItems: 'center',
		}}
	>
		<MovieCard {...args} />
	</div>
);

export const Default = Template.bind({});
Default.args = {
	movie: movies.popular[0],
};
