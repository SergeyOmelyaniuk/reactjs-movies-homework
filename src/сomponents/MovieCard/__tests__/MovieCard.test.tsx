import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import MovieCard from '..';
import { movies } from '../../../mocks/movies';

describe('MovieCard', () => {
	test('should render', () => {
		const { container } = render(
			<BrowserRouter>
				<MovieCard movie={movies.popular[0]} />
			</BrowserRouter>
		);

		expect(container).toMatchSnapshot();
	});
});
