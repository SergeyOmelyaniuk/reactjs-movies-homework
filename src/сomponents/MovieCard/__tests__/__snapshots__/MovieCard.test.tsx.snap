// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render 1`] = `
<div>
  <div
    class="movieCard"
  >
    <div
      class="wrapImage"
    >
      <div
        class="overlay"
      />
      <img
        alt=""
        class="image"
        src="https://image.tmdb.org/t/p/w500/film_1.jpg"
      />
      <span
        class="rating"
      >
        68
      </span>
      <div
        class="preview"
      />
    </div>
    <div
      class="content"
    >
      <h2
        class="title"
      >
        Красотка на взводе
      </h2>
      <p
        class="genre"
      />
    </div>
  </div>
</div>
`;
